### PR TITLE
Refactories

### PR DESCRIPTION
**Pull request by xian**
Here's a start at splitting up `ManifestFactory` into discrete concerns.

### CODE REVIEW

File: robolectric-annotations/src/main/java/org/robolectric/annotation/Config.java [hunk 1]
```diff
@@ -102,6 +105,20 @@
...
+   * @return Build type used for resource resolution.
+   */
+  String buildType() default DEFAULT_BUILD_TYPE;
+
+  /**
+   * Android product flavor to use for the resource resolution, such as "demo".
+   *
+   * @return Flavor used for resource resolution.
+   */
+  String flavor() default DEFAULT_FLAVOR;
```
Comment (JakeWharton) [inline robolectric-annotations/src/main/java/org/robolectric/annotation/Config.java]: This should probably be `String[]`, right? I can specify multiple flavors using dimensions: http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Multi-flavor-variants

Comment (xian) [inline robolectric-annotations/src/main/java/org/robolectric/annotation/Config.java]: Hey Jake!

Huh, interesting, didn't know about that. What would you think of using hyphen-separated `String flavors` (like `String qualifiers`), so we could reserve `String[] flavors(es?)` for multi test run configs, as sdk is used now?

Comment (JakeWharton) [inline robolectric-annotations/src/main/java/org/robolectric/annotation/Config.java]: works for me!

**Comment by xian** [comment]
Ended up taking `ManifestFactory` in a different direction.

**Pull request closed without merging**